src/model.jl

```diff
@@ -299,19 +291,12 @@ function _add_constraint_load_shed!(
         bus_demand[sets.load_bus_idx[i], j],
     )
     if demand_flexibility.enabled
-        if demand.flexibility.cost_input_type == "AREA"
-            demand_for_load_shed = JuMP.@expression(
-                m,
-                [i=1:sets.num_load_bus, j=1:interval_length],
-                demand_for_load_shed[i, j] + m[:load_shift_up][i, j] - m[:load_shift_dn][i, j],
-            )
-        elseif demand.flexibility.cost_input_type == "BUS"
-            demand_for_load_shed = JuMP.@expression(
-                m,
-                [i=1:sets.num_load_bus, j=1:interval_length],
-                demand_for_load_shed[i, j] + m[:load_shift_up][i, j] - m[:load_shift_dn][i, j],
-            )
-        end
+        demand_for_load_shed = JuMP.@expression(
+            m,
+            [i in sets.flexible_bus_idx, j=1:interval_length],
+            demand_for_load_shed[i, j] + m[:load_shift_up][i, j] - m[:load_shift_dn][i, j],
+        )
```

Comment:
I think we might need to rethink how we update this expression. I believe this expression overwrites the previous `demand_for_load_shed` expression, which is a problem when the set of flexible load buses is not equal to the set of load buses. When the set of flexible load buses is not equal to the set of load buses, this second iteration of `demand_for_load_shed` makes it so that there is only a value for buses that are included in the set of flexible load buses. This is inaccurate because `load_shed` occurs at each of the load buses (`load_shed` is not demand flexibility here, but rather a means to allow the simulation to avoid infeasibilities).

One solution that I can think of is doing something like we do in `_add_constraint_power_balance!`, where we could multiply the `load_shift_up` and `load_shift_dn` variables by some map between the load buses and the flexible load buses. Maybe something like this:

```
demand_for_load_shed = JuMP.@expression(
    m,
    demand_for_load_shed 
    + {map between load buses and flexible load buses} * m[:load_shift_up] 
    - {map between load buses and flexible load buses} * m[:load_shift_dn],
)
```

where `{map between load buses and flexible load buses}` would probably be created in a similar fashion to `flexible_load_bus_map` and may not need to be a part of `sets` (if this is the only place it is used).